pages/en/lb4/Creating-components.md

```diff
@@ -307,3 +307,297 @@ class EmailComponent {
   }
 }
 ```
+
+## Creating your own servers 
+
+LoopBack 4 has the concept of a Server, which you can use to create your own
+implementations of REST, SOAP, gRPC, MQTT and more. For an overview, see 
+[Server](Server.html).
+
+### A Basic Example
+The concept of servers in LoopBack doesn't actually require listening on ports;
+you can use this pattern to create workers that process data, write logs, or
+whatever you'd like.
+
+Let's create an example server! Please note that this is not a complete example:
+
+```ts
+import {Server} from '@loopback/core';
+import {processReports} from '../util';
+export class ReportProcessingServer implements Server {
+  @inject('database') db: DefaultCrudRepository<Report, int>;
+  interval: NodeJS.Timer;
+  async start() {
+    // Run the report processor every 2 minutes.
```

Comment:
Uh oh, I find this example rather complex and difficult to understand. Can we come up with a simpler example? If not, then please at least refactor the code and extract small named functions that will make it easier to comprehend what's going on. For example:

```ts
export class ReportProcessingServer implements Server {
  // ...
  async start() {
    // Run the report processor every 2 minutes.
    this.interval = setInterval(
      () => this.processReports().catch(err => console.warn('Report processing failed:', err)),
      1000 * 120);
  }

  async processReports() {
    const results = await this.getRecentlyAddedRecords();
    if (!results || !results.length) return;
    await Promise.all(results.map(rep => this.processResult(rep)));
  }

  // ...
}
```